### DISCUSSION TITLE
Issue with vernemq/vernemq:1.9.2 docker image with RHEL 9.x

### DISCUSSION THREAD

**Discussion by Krishnamotive** [vernemq/vernemq#2337] [Q&A]
Unable to use vernemq/vernemq:1.9.2 with RHEL 9.x. once the docker image is run the memory and CPU gets exhausted. The same works well with RHEL 8.x. Could you please help what might be the reason?
The logs also doesn't get generated to identify the potential issue.

**Comment by ioolkos** [comment]
@Krishnamotive Thanks, since `1.9.2` is from 2019, we cannot actively help/investigate this.

Why not try one of the newer packages? Incidentally, note that I have removed a bunch of older images from Docker Hub. Even if you *want* to stay on `1.9.2` you might want to see whether it compiles on a newer OTP release, or investigate whether some OS logs indicate any issues (maybe regarding OpenSSL versions etc)

**Comment by mths1** [comment]
My guess would be that you are running into the LimitNoFiles issue: https://docs.vernemq.com/guides/change-open-file-limits

**Reply by ioolkos** [reply]
Oh, that's a good point that I tend to forget. On RHEL 9, you cannot set LimitNoFiles to `infinity`.
OTOH, the Docker image does not use systemd integration at all, so there must be another issue at play.

**Comment by Krishnamotive** [comment]
Thank you for the revert
LimitNOFILE is already set a default to around a billion. Other parameters are tuned accordingly.

[cloud-user@hdm24 vernemq]$ systemctl show your-service --property=LimitNOFILE
LimitNOFILE=1073741816


There are no indications from the OS message log. The kernel simply points to the OOM killer freeing up system memory.

Aug  1 10:13:12 hdm24 systemd[1]: run-docker-runtime\x2drunc-moby-8605187143c300801862dd960e727d6c62a8fd2342d83edd04abf2f3d82fde98-runc.yPTFLr.mount: Deactivated successfully.
Aug  1 10:14:12 hdm24 systemd[1]: run-docker-runtime\x2drunc-moby-8605187143c300801862dd960e727d6c62a8fd2342d83edd04abf2f3d82fde98-runc.vASi84.mount: Deactivated successfully.

Aug  1 10:16:15 hdm24 kernel: beam.smp invoked oom-killer: gfp_mask=0xcc0(GFP_KERNEL), order=0, oom_score_adj=0
Aug  1 10:16:15 hdm24 kernel: CPU: 2 PID: 18382 Comm: beam.smp Not tainted 5.14.0-427.28.1.el9_4.x86_64 #1
Aug  1 10:16:15 hdm24 kernel: Hardware name: RDO OpenStack Compute, BIOS 1.11.0-2.el7 04/01/2014


Aug  1 10:16:15 hdm24 kernel: memory: usage 4194304kB, limit 4194304kB, failcnt 123384
Aug  1 10:16:15 hdm24 kernel: swap: usage 3144696kB, limit 4194304kB, failcnt 0


a8fd2342d83edd04abf2f3d82fde98.scope,mems_allowed=0,oom_memcg=/system.slice/docker-8605187143c300801862dd960e727d6c62a8fd2342d83edd04abf2f3d82fde98.scope,task_memcg=/system.slice/docker-8605187143c300801862dd960e727d6c62a8fd2342d83edd04abf2f3d82fde98.scope,task=beam.smp,pid=18299,uid=10000
Aug  1 10:17:15 hdm24 kernel: Memory cgroup out of memory: Killed process 18299 (beam.smp) total-vm:4871408kB, anon-rss:1482176kB, file-rss:4992kB, shmem-rss:0kB, UID:10000 pgtables:3396kB oom_score_adj:0
Aug  1 10:17:15 hdm24 kernel: beam.smp invoked oom-killer: gfp_mask=0xcc0(GFP_KERNEL), order=0, oom_score_adj=0
Aug  1 10:17:15 hdm24 kernel: CPU: 4 PID: 18466 Comm: beam.smp Not tainted 5.14.0-427.28.1.el9_4.x86_64 #1
Aug  1 10:17:15 hdm24 kernel: Hardware name: RDO OpenStack Compute, BIOS 1.11.0-2.el7 04/01/2014


I have picked up the latest version, which does not cause any issues. However, I needed to understand what is causing the problem with the old version since most of our testing was done on that version.
